compiler/rustc_ast_lowering/src/delegation.rs

```diff
@@ -55,9 +55,9 @@ use rustc_target::spec::abi;
 use std::iter;
 
 pub(crate) struct DelegationResults<'hir> {
-    pub body_id: hir::BodyId,
-    pub sig: hir::FnSig<'hir>,
-    pub generics: &'hir hir::Generics<'hir>,
+    pub(crate) body_id: hir::BodyId,
```

Comment:
Since the struct itself is `pub(crate)`, this doesn't really have any effects, right? It seems to just add more visual noise to me.

Reply:
> I don't think this lint should be applied to rustc until it's changed to not warn on `pub` fields of `pub(restricted)` structs

I posted https://github.com/rust-lang/rust/pull/126040 to address this.

Reply:
https://github.com/rust-lang/rust/pull/126040 was merged, the lint no longer triggers on struct fields.